api-server/api-server-common/src/storage/impls/in_memory/transactional/mod.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::{RwLock, RwLockReadGuard, RwLockWriteGuard};
+
+use common::chain::ChainConfig;
+
+use crate::storage::storage_api::{
+    ApiStorage, ApiStorageError, ApiTransactionRo, ApiTransactionRw, Transactional,
+};
+
+use super::ApiInMemoryStorage;
+
+pub mod read;
+pub mod write;
+
+pub struct ApiInMemoryStorageTransactionalRo<'t> {
+    transaction: RwLockReadGuard<'t, ApiInMemoryStorage>,
+}
+
+impl<'t> ApiInMemoryStorageTransactionalRo<'t> {
+    pub fn new(storage: &'t ThreadSafeApiInMemoryStorage) -> Self {
```

Comment:
This constructor (and similar for `Rw`) can be private. It's still accessible through the `Transactional` trait. Probably better to have a canonical way of creating these objects.

Reply:
Done.